### ISSUE TITLE
req.query does not support multiple values for same key

### ISSUE THREAD

**Issue by MattIPv4** [honojs/hono#257]
👋 If I make a request that uses the same query param key more than once, it doesn't seem like Hono has a way to handle that?

Say I make a request with `?test=1&test=2&test=3`, then `req.query('test')` will be `1` and `req.query().test` will be `3`.

I believe there should be a way to get all values associated with a key, similar to how `getAll` works for `URLSearchParams`. (`new URLSearchParams('test=1&test=2&test=3').getAll('test')` returns an array of `['1', '2', '3']`).

**Comment by MattIPv4** [comment]
For now I'm using a bit of middleware to patch this:

```js
// Patch req.query
app.use('*', async (ctx, next) => {
    const params = new URL(ctx.req.url).searchParams;
    ctx.req.query = (field = undefined) => field
        ? params.getAll(field)
        : [...params.entries()].reduce((obj, [ key, value ]) => ({
            ...obj,
            [key]: obj[key]
                ? (Array.isArray(obj[key]) ? obj[key] : [obj[key]]).concat([ value ])
                : value,
        }), {});
    await next();
});
```

**Comment by yusukebe** [comment]
Hi @MattIPv4 

I've implemented `c.req.queries` method in this PR #260 

In the handler, if you write:

```ts
const tests = req.queries('test')
```

and request to the endpoint:

```http
GET /?test=1&test=2&test=3 
```

you will get `tests` is `['1', '2', '3']`.

The next release will include this feature. Thank you!

**Comment by yusukebe** [comment]
@MattIPv4 

Hi! I've released **v1.3.6**. Please upgrade Hono!

**Comment by MattIPv4** [comment]
Thank you! 💙  Just updated and this is behaving perfectly.
